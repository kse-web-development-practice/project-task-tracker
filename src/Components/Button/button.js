import React from 'react'
import PropTypes from 'prop-types'
import styles from './button.module.css'

export const Button = ({ children, onClick, isMain }) => {
    return <button className={`${styles.button} ${isMain ? styles.main : ''}`} onClick={onClick}>{children}</button>
}

Button.propTypes = {
    children: PropTypes.node.isRequired,
    onClick: PropTypes.func,
    isMain: PropTypes.bool
}

Button.defaultProps = {
    isMain: false
}