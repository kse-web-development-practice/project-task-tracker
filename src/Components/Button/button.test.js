import React from 'react'
import { render } from '@testing-library/react'
import { Button } from './button'

describe('Button Component', () => {
  test('isMain is true', () => {
    const { asFragment } = render(<Button isMain>Click Me</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  test('isMain is false', () => {
    const { asFragment } = render(<Button isMain={false}>Click Me</Button>)
    expect(asFragment()).toMatchSnapshot()
  })
})
