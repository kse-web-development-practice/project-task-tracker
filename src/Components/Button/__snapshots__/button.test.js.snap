// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component isMain is false 1`] = `
<DocumentFragment>
  <button
    class="button  "
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button Component isMain is true 1`] = `
<DocumentFragment>
  <button
    class="button main "
  >
    Click Me
  </button>
</DocumentFragment>
`;
